test/unittests/utf8_test.cpp

```diff
@@ -0,0 +1,130 @@
+#include "utf8.hpp"
+#include <gtest/gtest.h>
+#include <test/utils/asserts.hpp>
+#include <test/utils/hex.hpp>
+
+using namespace fizzy;
+
+namespace
+{
+inline bool utf8_validate(bytes_view input)
+{
+    return fizzy::utf8_validate(input.begin(), input.end());
+}
+}  // namespace
+
+TEST(utf8, invalid_first_bytes)
+{
+    std::vector<bytes> testcases;
+    for (uint8_t c = 0x80; c < 0xC2; c++)
+        testcases.push_back(bytes(1, c));
+    for (uint8_t c = 0xF5; c < 0xFF; c++)
+        testcases.push_back(bytes(1, c));
+    testcases.push_back("ff"_bytes);
+    for (auto const& testcase : testcases)
```

Comment:
```suggestion
    for (const auto& testcase : testcases)
```

Reply:
Can you get this into the lint config?

Reply:
Btw copied from `leb128_test` which uses `auto const&`.

Reply:
I think clang-format is just working on it: https://reviews.llvm.org/D69764.